sdk/src/main/java/com/mapbox/maps/renderer/widget/BitmapWidgetRenderer.kt

```diff
@@ -80,12 +80,20 @@ internal class BitmapWidgetRenderer(
 
   private fun updateVertexBuffer() {
     // in pixels, (-bitmapWidth / 2, -bitmapHeight/2) - (bitmapWidth / 2, bitmapHeight/2)
-    vertexPositionBuffer.put(
-      -bitmapWidth / 2f, -bitmapHeight / 2f,
-      -bitmapWidth / 2f, bitmapHeight / 2f,
-      bitmapWidth / 2f, -bitmapHeight / 2f,
-      bitmapWidth / 2f, bitmapHeight / 2f,
-    )
+    when (position.horizontal) {
+      WidgetPosition.Horizontal.CENTER -> vertexPositionBuffer.put(
+        0f, 0f,
+        0f, bitmapHeight.toFloat(),
+        bitmapWidth.toFloat(), 0f,
+        bitmapWidth.toFloat(), bitmapHeight.toFloat()
+      )
+      else -> vertexPositionBuffer.put(
```

Comment:
I know you didn't write this code :slightly_smiling_face:, but maybe optimize and calculate `bitmapWidth / 2f` and `bitmapHeight / 2f` only once ?

```
val halfOfBitmapWidth = bitmapWidth / 2f
val halfOfBitmapHeight = bitmapHeight / 2f
vertexPositionBuffer.put(
   -halfOfBitmapWidth, -halfOfBitmapHeight,
   -halfOfBitmapWidth, -halfOfBitmapHeight,
   halfOfBitmapWidth, halfOfBitmapHeight,
   halfOfBitmapWidth, halfOfBitmapHeight
)
```

Reply:
I actually did so in my test branch but not committed here, will optimise once we have approved the solution :+1

Reply:
hard to say without testing but updating `translateMatrix` should be sufficient when bitmap size changed as the result of an update. vertex buffer looks correct to me as it is now

Reply:
Cleaner solution found by modifying topY and leftX methods, when centered left offset was applied twice.